flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarOperatorsTest.scala

```diff
@@ -62,6 +62,26 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
     )
   }
 
+  @Test
+  def testCast(): Unit = {
+
+    // binary -> varchar
+    testSqlApi(
+      "CAST (f18 as varchar)",
+      "hello world")
+
+    // varbinary -> varchar
+    testSqlApi(
+      "CAST (f19 as varchar)",
+      "hello flink")
+
+    // null case
+    testSqlApi("CAST (NULL AS INT)", "null")
+    testSqlApi(
+      "CAST (NULL AS VARCHAR) = ''",
+      "null")
```

Comment:
Would be better to add a test for cast binary literal to varchar.

Reply:
Updated.